compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -337,6 +337,11 @@ macro_rules! define_queries {
                 } else {
                     Some(key.default_span(*tcx))
                 };
+                let def_id = key.key_as_def_id();
+                let def_kind = def_id.map(|def_id| {
+                    let def_kind = tcx.def_kind(def_id);
```

Comment:
I just realized something: since this code is executed as part of running a query, should I change it to

```suggestion
                    let def_kind = tcx.hir().opt_def_kind(def_id).unwrap();
```

in order to ensure that `def_kind` doesn't run a query that could send the compiler into a loop?

Reply:
Yeah, that makes sense. Good catch.